packages/unmock-core/src/index.ts

```diff
@@ -1,34 +1,91 @@
-import { IBackend, IUnmockOptions, IUnmockPackage } from "./interfaces";
-import { UnmockOptions } from "./options";
+import { escapeRegExp } from "lodash";
+import { IBackend, ILogger, IUnmockPackage } from "./interfaces";
 import * as transformers from "./service/state/transformers";
 // top-level exports
-export { UnmockOptions } from "./options";
 export * from "./interfaces";
 export * from "./generator";
 export const dsl = transformers;
 
+const whitelistToRegex = (whitelist?: Array<string | RegExp>): RegExp[] =>
+  whitelist === undefined
+    ? []
+    : whitelist.map((item: any) =>
+        item instanceof RegExp
+          ? item
+          : new RegExp(
+              `^${item
+                .split(/\*+/)
+                .map(escapeRegExp)
+                .join(".*")}$`,
+            ),
+      );
+
 export abstract class CorePackage implements IUnmockPackage {
   protected readonly backend: IBackend;
-  private readonly options: UnmockOptions;
+  private logger: ILogger = { log: () => undefined }; // Default logger does nothing
+  private whitelistAsStrings: Array<string | RegExp> = [
+    "127.0.0.1",
+    "127.0.0.0",
+    "localhost",
+  ];
+  private regexWhitelist: RegExp[];
+  private useInProduction: boolean = false;
+
+  constructor(
+    backend: IBackend,
+    options?: {
+      logger?: ILogger;
+      whitelist?: string[] | string;
+      useInProduction?: boolean;
+    },
+  ) {
+    this.whitelistAsStrings =
+      options && options.whitelist
+        ? Array.isArray(options.whitelist)
+          ? options.whitelist
+          : [options.whitelist]
+        : this.whitelistAsStrings;
+    this.regexWhitelist = whitelistToRegex(this.whitelistAsStrings);
 
-  constructor(baseOptions: UnmockOptions, backend: IBackend) {
-    this.options = baseOptions;
     this.backend = backend;
+    this.logger = (options && options.logger) || this.logger;
+    this.useInProduction =
+      (options && options.useInProduction) || this.useInProduction;
   }
 
-  public on(maybeOptions?: IUnmockOptions) {
-    return this.backend.initialize(this.options.reset(maybeOptions));
+  public on() {
+    return this.backend.initialize({
+      useInProduction: this.useInProduction,
+      isWhitelisted: (url: string) => this.isWhitelisted(url),
+      log: (message: string) => this.logger.log(message),
+    });
   }
-  public init(maybeOptions?: IUnmockOptions) {
-    this.on(maybeOptions);
+  public init() {
+    this.on();
   }
-  public initialize(maybeOptions?: IUnmockOptions) {
-    this.on(maybeOptions);
+  public initialize() {
+    this.on();
   }
 
   public off() {
     this.backend.reset();
   }
 
+  public get whitelist() {
```

Comment:
Is `whitelist` the best name for this? Maybe something like `allowedHosts` or `passthroughHosts` would make the intention clearer 🤔